client/src/app/site/pages/meetings/pages/participants/pages/mandate-check/components/mandate-check-list/mandate-check-list.component.ts

```diff
@@ -0,0 +1,223 @@
+            this.updateEntries();
+            this.cd.markForCheck();
+        });
+
+        this.storage.get<Id[]>(`mandate-check-groups`).then(groups => this.form.get(`groups`).setValue(groups));
+    }
+
+    public override ngOnDestroy(): void {
+        super.ngOnDestroy();
+        this.storage.set(`mandate-check-groups`, this.selectedGroups);
+    }
+
+    public displayPercent(value: number): string {
+        if (Number.isNaN(value)) {
+            return ``;
+        }
+        return `(${Number(value * 100).toFixed(0)}%)`;
+    }
+
+    public toggle(structureLevelId: Id): void {
+        this.toggleMap.set(structureLevelId, !(this.toggleMap.get(structureLevelId) ?? true));
+    }
+
+    public getName(userId: Id): string {
+        return this.participantRepo.getViewModel(userId).getShortName();
+    }
+
+    private updateEntries(): void {
+        const filteredSortedParticipants = this.participants
+            .filter(pt => (this.selectedGroups ?? []).some(id => pt.group_ids().includes(id)))
+            .sort((a, b) => a.short_name.localeCompare(b.short_name));
+        this.genders = this.genderRepo.getViewModelList();
+        // female at first place
+        if (
+            this.genders.some(gender => gender.id === MALE_ID) &&
+            this.genders.some(gender => gender.id === FEMALE_ID)
+        ) {
+            this.genders = [this.genders[1], this.genders[0], ...this.genders.slice(2)];
+        }
+        this.genders.push({ id: UNKNOWN_ID, name: 'NA' });
+        const allMandates = new MandateCheckEntry(`All Mandates`, ALL_MANDATES_ID, this.genders);
+        const structureLevelsEntryMap = new Map<Id, MandateCheckEntry>();
+        for (const strLvl of this.structureLevels ?? []) {
+            structureLevelsEntryMap.set(strLvl.id, new MandateCheckEntry(strLvl.name, strLvl.id, this.genders));
+        }
+        for (const participant of filteredSortedParticipants) {
+            allMandates.add(participant.id, participant.isPresentInMeeting(), participant.gender_id);
+            for (const strLvlId of participant.structure_level_ids() ?? []) {
+                if (this.structureLevels.length) {
+                    structureLevelsEntryMap
+                        .get(strLvlId)
+                        .add(participant.id, participant.isPresentInMeeting(), participant.gender_id);
+                }
+            }
+        }
+        const sortedEntries = [...structureLevelsEntryMap.values()].sort((a, b) => a.name.localeCompare(b.name));
+        this.entries = [allMandates, ...sortedEntries];
+        this.entriesObservable.next([allMandates, ...sortedEntries]);
+    }
+
+    private updateToggleMap(): void {
+        const newMap = new Map<Id, boolean>();
+        newMap.set(ALL_MANDATES_ID, this.toggleMap.get(ALL_MANDATES_ID) ?? false);
+        for (const strlvl of this.structureLevels) {
+            newMap.set(strlvl.id, this.toggleMap.get(strlvl.id) ?? false);
```

Comment:
Any reason why you have to user `?? false` in these lines instead of `!!this.toggleMap.get(ALL_MANDATES_ID)` and `!!this.toggleMap.get(strlvl.id)`?